mantidproject/mantid
# Pass GroupingWorkspace information to CreateCacheFilename in AlignAndFocusPowderFromFiles

**Description of work.**
Pass GroupingWorkspace information to CreateCacheFilename in AlignAndFocusPowderFromFiles
**Report to:** [Marshall McDonnel]/[nobody]. 

**To test:**

Here is an example that shows that since the grouping workspace name is the same, the sha will be the same as well, thus the cache file will be used when it in fact should not be since the grouping has changed:

NOTE: Need access to NOMAD data to run.

```python
import os
from mantid.simpleapi import *

# cleanup
files = [os.path.join('/tmp', filename) for filename in os.listdir('/tmp')
         if filename.startswith('NOM') and filename.endswith('nxs')]
for filename in files:
    os.remove(filename)

# setup args
filenames = ['/SNS/NOM/IPTS-17210/nexus/NOM_77521.nxs.h5',
             '/SNS/NOM/IPTS-17210/nexus/NOM_77532.nxs.h5',
             '/SNS/NOM/IPTS-17210/nexus/NOM_77543.nxs.h5']

alignAndFocusArgs = dict(Filename=','.join(filenames),
                         GroupingWorkspace='grp_wksp',
                         MaxChunkSize=8,
                         CalFileName='/SNS/NOM/shared/CALIBRATION/2016_2_1B_CAL/NOM_d77540_2016_07_23_shifter.h5',
                         ResampleX=-6000,
                         TMin=300.0, TMax=16667.0,
                         CacheDir='/tmp')

# initially with all grouping
CreateGroupingWorkspace(InstrumentName="NOMAD", GroupDetectorsBy='All', OutputWorkspace='grp_wksp')
for outname in ['first', 'second']:
    AlignAndFocusPowderFromFiles(OutputWorkspace=outname,
                                 **alignAndFocusArgs)
    NormaliseByCurrent(InputWorkspace=outname, OutputWorkspace=outname,
                       RecalculatePCharge=True)

Minus(LHSWorkspace='first', RHSWorkspace='second', OutputWorkspace='diff')
import numpy as np
for i in range(mtd['first'].getNumberHistograms()):
    print('%d: x=%s y=%s' % (i, str(np.alltrue(mtd['first'].readX(i) == mtd['second'].readX(i))),
    str(np.alltrue(mtd['first'].readY(i) == mtd['second'].readY(i)))))
CompareWorkspaces(Workspace1='first', Workspace2='second',
                  CheckAxes=False)

# now with bank grouping
CreateGroupingWorkspace(InstrumentName="NOMAD", GroupDetectorsBy='bank', OutputWorkspace='grp_wksp')
for outname in ['second']:
    AlignAndFocusPowderFromFiles(OutputWorkspace=outname,
                                 **alignAndFocusArgs)
    NormaliseByCurrent(InputWorkspace=outname, OutputWorkspace=outname,
                       RecalculatePCharge=True)

#Should fail if new group was used
Minus(LHSWorkspace='first', RHSWorkspace='second', OutputWorkspace='diff')
```

Fixes #24074 



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
